common/dynamicconfig/constants.go

```diff
@@ -3454,6 +3468,21 @@ var IntKeys = map[IntKey]DynamicInt{
 		Description:  "The rate for which sampled logs are logged at. 100 means 1/100 is logged",
 		DefaultValue: 100,
 	},
+	LargeShardHistorySizeMetricThreshold: DynamicInt{
+		KeyName:      "system.largeShardHistorySizeMetricThreshold",
+		Description:  "defines the threshold for what consititutes a large history size to alert on, default is 10mb",
+		DefaultValue: 10485760,
+	},
+	LargeShardHistoryEventMetricThreshold: DynamicInt{
+		KeyName:      "system.largeShardHistoryEventMetricThreshold",
+		Description:  "defines the threshold for what consititutes a large history event length to alert on, default is 10000",
+		DefaultValue: 10000,
+	},
```

Comment:
is this a per-workflow-history-count threshold?  would it make sense to use the same default value as the warning level?

Reply:
I think because we've been telling people 10k in cadence users but default is 25k which seems to be from over 2 years ago. We can always change in flipr as needed.

Reply:
I mean more that this is dependent on the datastore / other config, no one value is "correct".  and our internal values are largely irrelevant.

10k seems probably reasonable, and I'm not sure if we can actually make this a "the default matches the dynamic value of [another key]" without a lot more work.  just wondering if it'd make more sense to be hardcoded as the same as another, to show the implied relationship.

Reply:
How about I set both to the same hard coded value initially but take the lesser of the 2 when it comes to metric emission?

Reply:
yeah i think this is a good idea since those values are actually configurable on a domain level so doing a emitMetricWarning = generalWarn<domainWarn?generalWarn:domainWarn seems good.

Reply:
yea, min of both in code makes sense I think.  seems fairly likely that user-warning-level also means approaching "big".

worst case, we just change our minds later 🤷